### PR TITLE
test(e2e,playwright): link-ms-user component

### PR DESCRIPTION
**Pull request by shabeebk**
With this commit, the playwright will test the  link-ms-user component for possible view for signedin/sighedout user.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref https://github.com/freeCodeCamp/freeCodeCamp/issues/51705

### CODE REVIEW

**Comment by shabeebk** [comment]
camperbot cant code

File: e2e/link-ms-user.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+test.beforeEach(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto(
+    '/learn/foundational-c-sharp-with-microsoft/write-your-first-code-using-c-sharp/trophy-write-your-first-code-using-c-sharp'
+  );
+});
+
+test.afterEach(async () => {
+  await page.close();
+});
```
Comment (naomi-lgbt) [inline e2e/link-ms-user.spec.ts:15]:
```suggestion
test.beforeEach(async ({ page }) => {
  await page.goto(
    '/learn/foundational-c-sharp-with-microsoft/write-your-first-code-using-c-sharp/trophy-write-your-first-code-using-c-sharp'
  );
});

test.afterEach(async ({ page }) => {
  await page.close();
});
```

File: e2e/link-ms-user.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { test, expect, type Page } from '@playwright/test';
+import translations from '../client/i18n/locales/english/translations.json';
+
+let page: Page;
```
Comment (naomi-lgbt) [inline e2e/link-ms-user.spec.ts:4]:
```suggestion
import { test, expect } from '@playwright/test';
import translations from '../client/i18n/locales/english/translations.json';
```

File: e2e/link-ms-user.spec.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    '/learn/foundational-c-sharp-with-microsoft/write-your-first-code-using-c-sharp/trophy-write-your-first-code-using-c-sharp'
+  );
+});
+
+test.afterEach(async () => {
+  await page.close();
+});
+
+test.describe('Link MS user component (unlinked signedOut user)', () => {
+  test('Component has proper main heading and relevant sections', async () => {
```
Comment (naomi-lgbt) [inline e2e/link-ms-user.spec.ts:18]:
```suggestion
  test('Component has proper main heading and relevant sections', async ({ page }) => {
```

File: e2e/link-ms-user.spec.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    const linkSignInText = page.getByTestId('link-signin-text');
+    await expect(linkSignInText).toBeVisible();
+  });
+});
+
+test.describe('Link MS user component (unlinked signedIn user)', () => {
+  test.use({ storageState: 'playwright/.auth/certified-user.json' });
+
+  test('Component has proper main heading and relevant sections', async () => {
```
Comment (naomi-lgbt) [inline e2e/link-ms-user.spec.ts:32]:
```suggestion
  test('Component has proper main heading and relevant sections', async ({ page }) => {
```

File: e2e/link-ms-user.spec.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    const mainHeading = page.getByRole('heading', {
+      name: translations.learn.ms['link-header']
+    });
+    await expect(mainHeading).toBeVisible();
+
+    const linkSignInText = page.getByTestId('unlinked-text');
+    await expect(linkSignInText).toBeVisible();
+  });
+
+  test('Component has proper list of actions', async () => {
```
Comment (naomi-lgbt) [inline e2e/link-ms-user.spec.ts:42]:
```suggestion
  test('Component has proper list of actions', async ({ page }) => {
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This is looking good. Shouldn't need a global page variable, though.

**Comment by shabeebk** [comment]
Adapted the suggestion and applied the commits.

**Review by shabeebk — COMMENTED** [review]
Accepted changes

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @shabeebk :tada: Thanks for contributing to freeCodeCamp and happy hacktoberfest :tada:

**Pull request merged**